jormungandr/src/topology/process.rs

```diff
@@ -0,0 +1,142 @@
+            reset_layers(internal_msgbox_1.clone())
+        });
+    }
+
+    // Send gossips for trusted peers at the beginning, without inserting them first in
+    // the topology
+    for peer in config.trusted_peers {
+        let gossips = topology.initiate_gossips(None);
+        network_msgbox
+            .send(NetworkMsg::Propagate(PropagateMsg::Gossip(
+                Peer {
+                    addr: peer.addr,
+                    id: peer.id,
+                },
+                gossips,
+            )))
+            .await
+            .unwrap_or_else(|e| tracing::error!("Error sending gossips to network task: {}", e));
+    }
+
+    info.run_periodic_fallible(
+        "send gossips to network task to propagate",
+        config.gossip_interval,
+        move || handle_gossip(internal_msgbox_2.clone(), network_msgbox.clone()),
+    );
+
+    let mut process = Process { input, topology };
+
+    process.handle_input().await;
+}
+
+impl<T: Stream<Item = Message> + Unpin> Process<T> {
+    async fn handle_input(&mut self) {
+        use Message::{Private, Public};
+        while let Some(input) = self.input.next().await {
+            match input {
+                Public(TopologyMsg::AcceptGossip(gossip)) => self.topology.accept_gossips(gossip),
+                Public(TopologyMsg::DemotePeer(id)) => self.topology.report_node(&id),
+                Public(TopologyMsg::PromotePeer(id)) => self.topology.promote_node(&id),
+                Public(TopologyMsg::View(selection, handle)) => {
+                    handle.reply_ok(self.topology.view(selection))
+                }
+                Public(TopologyMsg::ListAvailable(handle)) => {
+                    handle.reply_ok(self.topology.list_available())
+                }
+                Public(TopologyMsg::ListNonPublic(handle)) => {
+                    handle.reply_ok(self.topology.list_non_public())
+                }
+                Public(TopologyMsg::ListQuarantined(handle)) => {
+                    handle.reply_ok(self.topology.list_quarantined())
+                }
+                Private(PrivateMsg::ResetTopology) => self.topology.force_reset_layers(),
+                Private(PrivateMsg::GetGossips(handle)) => {
+                    let view = self.topology.view(poldercast::layer::Selection::Any);
+                    let mut res = Vec::new();
+                    for peer in view.peers {
+                        // Peers returned bu the topology will always have a NodeId
+                        let id = peer.id.clone().unwrap();
+                        res.push((peer, self.topology.initiate_gossips(Some(&id))));
+                    }
+                    handle.reply_ok(res);
+                }
+            }
+        }
+    }
+}
+
+async fn reset_layers(
+    mut mbox: MessageBox<PrivateMsg>,
+) -> Result<(), futures::channel::mpsc::SendError> {
+    mbox.send(PrivateMsg::ResetTopology).await
+}
```

Comment:
Do you mean the `reset_layers` function or the whole private message box?
As for the first, we can remove it since we've decided we won't support that function.
The private message box was meant to serve as a sort of a task private interface for the topology input handling loop. 
We can't pass the topology struct around to other functions, so I thought of adding `Private` messages that are used for things that should be handled by the topology task (i.e.  periodic gossiping) but I would rather put somewhere else than in the input loop.

Reply:
Maybe just the periodic interval source that could be pulled together with the public queue?